core/node/commitment_generator/src/utils.rs

```diff
@@ -234,3 +241,79 @@ pub(crate) fn convert_vm_events_to_log_queries(events: &[VmEvent]) -> Vec<LogQue
         })
         .collect()
 }
+
+pub(crate) fn pubdata_to_blob_linear_hashes(
+    blobs_required: usize,
+    mut pubdata_input: Vec<u8>,
+) -> Vec<H256> {
+    // Now, we need to calculate the linear hashes of the blobs.
+    // Firstly, let's pad the pubdata to the size of the blob.
+    if pubdata_input.len() % ZK_SYNC_BYTES_PER_BLOB != 0 {
+        let padding =
+            vec![0u8; ZK_SYNC_BYTES_PER_BLOB - pubdata_input.len() % ZK_SYNC_BYTES_PER_BLOB];
+        pubdata_input.extend(padding);
```

Comment:
Nit: There's no need to allocate an additional `Vec` here; it's possible to use [`Vec::resize()`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.resize).